code/game/objects/items/weapons/RTD.dm

```diff
@@ -0,0 +1,594 @@
+Can rapidly collect iron ore to be used as matter.
+Holds 400 matter!
+EMAGGED FUNCTIONS - TODO
+*/
+/obj/item/weapon/rtd
+	name = "rapid-terraforming device (RTD)"
+	desc = "A bulky, prototype industrial device based on RCD and compressed matter technology that is designed for rapid terraforming of hostile planetary environments. \
+	Has safeguards to not work on station. Automatically synthesizes compressed air to pressurize colonies. A label on the side reads: WARNING: DO NOT USE DEVICE WITHOUT INSULATED GLOVES!"
+	icon = 'icons/obj/tools.dmi'
+	icon_state = "rtd"
+	opacity = 0
+	density = 0
+	anchored = 0
+	flags = CONDUCT
+	force = 15
+	throwforce = 10
+	throw_speed = 3
+	throw_range = 5
+	w_class = 4
+	materials = list(MAT_METAL=100000)
+	origin_tech = "engineering=5;materials=5"
+	armor = list(melee = 0, bullet = 0, laser = 0, energy = 0, bomb = 0, bio = 0, rad = 0, fire = 100, acid = 100)
+	resistance_flags = FIRE_PROOF | ACID_PROOF
+	var/datum/effect_system/spark_spread/spark_system
+	var/matter = 400
+	var/maxmatter = 400
+	var/gas = 10
+	var/maxgas = 10
+	var/working = 0
+	var/mode = 1
+	var/canRturf = 0
+	var/airlock_type = /obj/machinery/door/airlock
+	var/range_catwalk = 4
+	var/range_plating = 3
+	var/o2ratio = 0.2
+	var/n2ratio = 0.8
+	var/gas_use = 1
+
+	var/advanced_airlock_setting = 1
+	var/sheetmultiplier	= 8
+	var/plasteelmultiplier = 5
+	var/ironoreworth = 5
+	var/gas_amount = (MOLES_CELLSTANDARD * 2)
+	var/gas_regen_delay = 5
+
+	var/list/conf_access = null
+	var/use_one_access = 0 //If the airlock should require ALL or only ONE of the listed accesses.
+
+	var/no_ammo_message = "Insufficient Matter."
+	var/safety = 1
+	/*
+	Construction costs
+	COST| FUNCTION
+	10	|Asteroid walls to walls
+	5	|Lavaland floors to plating
+	5	|Builds catwalks on lava.
+	SEPARATE FROM OTHERS - Blasts compressed air to pressurize areas
+	25	|Builds airlocks with higher health than normal airlocks, but the same security level.
+	15	|Catwalks over chasms
+	2	|Build membrane barriers - Basically walls that will break in a single hit from just about anything and is best left for emergency usage.
+	0	|Remove Catwalks - Just incase antags get ahold of it or you really need a lava disposals bin. Honk.
+	*/
+
+	var/wallcost = 10
+	var/floorcost = 5
+	var/catwalkcost = 5
+	var/chasmcatwalkcost = 15
+	var/airlockcost = 25
+	var/membranecost = 2
+	var/removecatwalkcost = 0
+	var/processtick = 0
+
+	/* Build delays (deciseconds) */
+
+	var/walldelay = 20
+	var/floordelay = 10
+	var/catwalkdelay = null
+	var/chasmcatwalkdelay = null
+	var/airlockdelay = 30
+	var/membranedelay = 5
+	var/removedelay = 40
+
+/obj/item/weapon/rtd/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] sets the RTD to 'Airlock' and points it down [user.p_their()] throat! It looks like [user.p_theyre()] trying to commit suicide..</span>")
+	var/obj/machinery/door/airlock/A = new /obj/machinery/door/airlock/glass_command(get_turf(src))
```

Comment:
@Dagdammit btw it has the standard RCD settings so really it can do it anyways, its just in this situation the airlock has 500 health.
also, it's a suicide_act thing like nuke disks, I don't see an issue but I'l change it if people think it's a problem?